### DISCUSSION TITLE
Overriding opentelemetry TraceExporter configuration

### DISCUSSION THREAD

**Discussion by t0xa** [quarkusio/quarkus#29143] [Q&A] [answered]
Hello all,

Problem statement:
- I'm deploying quarkus application to Google Cloud Platform (GCP)
- I would like to use Cloud Trace service. Cloud Trace supports Opentelemetry (https://cloud.google.com/trace/docs/setup/java-ot)
- Google provides library `com.google.cloud.opentelemetry` that takes care of configuring `TraceExporter` for use in GCP platform.
- I cannot find a way to override `TraceExport` configuration in `quarkus-opentelemetry` and `quarkus-opentelemetry-exporter-otlp`

Could you please point me to right direction?

Thank you in advance

[a comment by an automated account is not shown]

**Comment by brunobat** [comment, marked as answer]
Hi @t0xa.
You can try to create a CDI bean producer for the [TraceExporter](https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/main/exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java). Like in here:
https://github.com/quarkusio/quarkus/blob/main/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpExporterProvider.java
Ideally, we should create a new exporter in quarkiverse here: https://github.com/quarkiverse/quarkus-opentelemetry-exporter
This way, we would only need to include the dependency.

**Comment by t0xa** [comment]
CDI Producer worked for me. Thanks for your help @brunobat
